#168377676 Refactoring routes code

gadishimwe opened the pull request:
#### What does this PR do?
Refactoring routes code
#### Description of Task to be completed?
- Created server.js as an entry point
- Made index.js a middleware to other routes

#### How should this be manually tested?
By running ```npm run dev``` or ```npm run test```
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[Finishes #168377676]

#### Screenshots
![Screenshot at 2019-09-09 16-35-38](https://user-images.githubusercontent.com/52411901/64540524-a7414e80-d320-11e9-9b0e-6cb04aaf5e73.png)

#### Questions
N/A

Commit:
Chore(Refactoring): Refactoring routes code [Finishes #168377676]

Oluwasegun-AA commented:
@gadishimwe , please rename this to `routes` . 
![image](https://user-images.githubusercontent.com/25525765/64551811-3b69e080-d336-11e9-950c-8ecfb1e4d5cb.png)

Oluwasegun-AA commented on server.js:
```diff
@@ -1,31 +1,22 @@
-app.use('/api/v1/', mentor);
-app.use('/api/v1/', session);
+app.use('/', index);
 app.get('/', (req, res) => {
```
@gadishimwe , this routing should be implemented in your `index.js` file in the `routes` folder

Oluwasegun-AA commented on server/routes/index.js:
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable no-console */
```
@gadishimwe , do not disable any Eslint flag. the purpose of using the Airbnb configuration is to ensure that we write our code in line with a predefined standard. kindly remove all Eslint disabling comments and adjust the codes accordingly

Commit:
Chore(refactoring routes): refactoring routes code

- Rename index to routes in server.js
- Move welcome message route to index.js in routes
- Re-enable Eslint flags and corrected the mistakes flaged by Eslint

[Finishes #168377676]
